data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/S3ScanPartitionCreationSupplier.java

```diff
@@ -186,9 +191,10 @@ private void initializeGlobalStateMap(final Map<String, Object> globalStateMap)
         globalStateMap.put(SINGLE_SCAN_COMPLETE, false);
     }
 
-    private boolean isLastModifiedTimeAfterMostRecentScanForBucket(final Instant previousScanTime,
+    @VisibleForTesting
+    boolean isLastModifiedTimeAfterMostRecentScanForBucket(final Instant previousScanTime,
                                                                    final S3Object s3Object) {
-        if (previousScanTime == null) {
+        if (previousScanTime == null || deleteS3ObjectsOnRead) {
```

Comment:
QUES: could it be confusing to put deleteS3ObjectsOnRead under the function context? Is it better to directly put it in the filter?

Reply:
It seems fine either way as it is directly relevant to this filter, but it could go in the filter directly